### PR TITLE
[C#] Namespace concept. Red vs Blue. Tests

### LINKED ISSUE #2026: [C#] namespaces - enable this to build out-of-the-box
Currently, the project will not build until a `Combined` namespace has been created. This is not satisfactory.

Perhaps we have to use reflection to find the namespaces.

### PR DESCRIPTION
**Pull request by valentin-p**
- Removed using Combined from Tests
- Added tests based on the type

I did not add any unit tests for the aliases Red/Blue.

Closes #2026

### CODE REVIEW

**Review by mikedamay — COMMENTED** [review]
This looks good.

I was not sure why the warnings were disabled.  I did not see them on my build when I removed the pragma.  If we can't get rid of them I suppose we will need to introduce the compiler directive concept.

@ErikSchierboom what do you think?

**Comment by valentin-p** [comment]
We can disable pragma in the project file too if necessary.

File: languages/csharp/exercises/concept/red-vs-blue-darwin-style/RedVsBlueDarwinStyleTests.cs [hunk 1]
```diff
@@ -1,17 +1,64 @@
...
+        var carBuilderType = Type.GetType(GetNamespaceDotClass(CombinedNamespace, CarBuilderClass));
+        object redRemoteControlCar = carBuilderType?.GetMethod(BuildRedMethod)?.Invoke(null, null);
+        Assert.NotNull(redRemoteControlCar);
+    }
+
+    private const string CombinedNamespace = "Combined";
+    private const string CarBuilderClass = "CarBuilder";
+    private const string BuildRedMethod = "BuildRed";
+    private const string BuildBlueMethod = "BuildBlue";
+    private string GetNamespaceDotClass(string @namespace, string @class) => $"{@namespace}.{@class}";
```
Comment (ErikSchierboom) [inline languages/csharp/exercises/concept/red-vs-blue-darwin-style/RedVsBlueDarwinStyleTests.cs:63]: Could you inline the constants and the method call? This makes it far easier for the test-runner to generate its output (see https://github.com/exercism/csharp-test-runner/pull/9).

File: languages/csharp/exercises/concept/red-vs-blue-darwin-style/RedVsBlueDarwinStyle.cs [hunk 2]
```diff
@@ -1,3 +1,5 @@
+#pragma warning disable RCS1163, RCS1110, IDE0060
+
```
Comment (ErikSchierboom) [inline languages/csharp/exercises/concept/red-vs-blue-darwin-style/RedVsBlueDarwinStyle.cs:2]: Like Mike, I've also checked if I get any warnings when removing this line, and for me it also doesn't result in any difference. Could you remove it?

File: languages/csharp/exercises/concept/red-vs-blue-darwin-style/.meta/Example.cs [hunk 3]
```diff
@@ -1,3 +1,5 @@
+#pragma warning disable RCS1163, RCS1110, IDE0060
+
```
Comment (ErikSchierboom) [inline languages/csharp/exercises/concept/red-vs-blue-darwin-style/.meta/Example.cs:2]: I don't think the example needs this, right?

**Review by ErikSchierboom — APPROVED** [review]
Great work! I've left some comments

File: languages/csharp/exercises/concept/red-vs-blue-darwin-style/RedVsBlueDarwinStyleTests.cs [hunk 4]
```diff
@@ -1,17 +1,64 @@
...
+        var carBuilderType = Type.GetType(GetNamespaceDotClass(CombinedNamespace, CarBuilderClass));
+        var returnType = carBuilderType?.GetMethod(BuildRedMethod)?.ReturnType;
+        Assert.Equal(GetNamespaceDotClass("RedRemoteControlCarTeam", "RemoteControlCar"), returnType?.FullName);
+    }
+
+    [Fact(Skip = "Remove this Skip property to run this test")]
+    public void Namespace_for_CarBuilder_can_BuildBlue_car()
+    {
+        var carBuilderType = Type.GetType(GetNamespaceDotClass(CombinedNamespace, CarBuilderClass));
+        object blueRemoteControlCar = carBuilderType?.GetMethod(BuildBlueMethod)?.Invoke(null, null);
```
Comment (ErikSchierboom) [inline languages/csharp/exercises/concept/red-vs-blue-darwin-style/RedVsBlueDarwinStyleTests.cs:47]:
```suggestion
        var blueRemoteControlCar = carBuilderType?.GetMethod(BuildBlueMethod)?.Invoke(null, null);
```

File: languages/csharp/exercises/concept/red-vs-blue-darwin-style/RedVsBlueDarwinStyleTests.cs [hunk 5]
```diff
@@ -1,17 +1,64 @@
...
+        Assert.NotNull(blueRemoteControlCar);
+    }
+
+    [Fact(Skip = "Remove this Skip property to run this test")]
+    public void Namespace_for_CarBuilder_can_BuildRed_car()
+    {
+        var carBuilderType = Type.GetType(GetNamespaceDotClass(CombinedNamespace, CarBuilderClass));
+        object redRemoteControlCar = carBuilderType?.GetMethod(BuildRedMethod)?.Invoke(null, null);
```
Comment (ErikSchierboom) [inline languages/csharp/exercises/concept/red-vs-blue-darwin-style/RedVsBlueDarwinStyleTests.cs:55]:
```suggestion
        var redRemoteControlCar = carBuilderType?.GetMethod(BuildRedMethod)?.Invoke(null, null);
```

**Comment by mikedamay** [comment]
I did not really understand the inline issue so @ErikSchierboom can you review the changes.

**Review by ErikSchierboom — APPROVED** [review]

**Comment by ErikSchierboom** [comment]
Thanks @valentin-p! 🎉

**Pull request merged**